SweetsFactory/src/app/category/category.component.ts

```diff
@@ -1,17 +1,45 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import { CategoryService } from '../shared/services/category.service';
 import { Stock } from '../shared/model/stock';
+import { Category } from '../shared/model/category';
 
 @Component({
     templateUrl: './category.component.html',
     styleUrls: ['./category.component.css']
 })
-export class CategoryComponent {
+export class CategoryComponent implements OnInit {
     stocks: Stock[];
+    emptyCategoryName = false;
+    categoryName: string;
     header: string = "Categories";
+    reveal: any;
+
+    ngOnInit(): void {
+        this.reveal = $('.reveal');
+        this.reveal.foundation();
+    }
 
     constructor(private categoryService: CategoryService) {
         this.stocks = categoryService.stock;
+        categoryService.categoryAnnounced$.subscribe(
+            newCategories => {
+                this.stocks = categoryService.stock;
+            }
+        )
+    }
+
+    saveNewCategory() {
+        if (this.categoryName !== undefined && this.categoryName !== ' ') {
+            this.categoryService.saveCategory(this.categoryName);
+            this.reveal.foundation('close');
+            this.categoryName = '';
+        } else {
+            this.emptyCategoryName = true;
```

Comment:
id the category name can be undefined or empty, why do you need an extra field "emptyCategoryName" ?